core-experimental/include/iDynTree/Core/Rotation.h

```diff
@@ -42,7 +42,12 @@ namespace iDynTree
     {
     private:
         RotationSemantics semantics;
-
+        
+        /**
+         * Copy constructor: create a Rotation from another RotationRaw and another RotationSemantics.
+         */
+        Rotation(const RotationRaw & other, RotationSemantics & semantics);
```

Comment:
This constructor is only for internal use in semantic operations of Rotation class (`compose`, `inverse2`,...)